src/Dompdf.php

```diff
@@ -727,24 +728,38 @@ public function render()
 
         // @page style rules : size, margins
         $pageStyles = $this->css->get_page_styles();
-
         $basePageStyle = $pageStyles["base"];
         unset($pageStyles["base"]);
 
         foreach ($pageStyles as $pageStyle) {
             $pageStyle->inherit($basePageStyle);
         }
 
+        $defaultOptionPaperSize = $this->getPaperSize($options->getDefaultPaperSize());
+        // If there is a CSS defined paper size compare to the paper size used to create the canvas to determine a
+        // recreation need
         if (is_array($basePageStyle->size)) {
-            $this->setPaper(array(0, 0, $basePageStyle->size[0], $basePageStyle->size[1]));
+            $basePageStyleSize = $basePageStyle->size;
+            if ($defaultOptionPaperSize[0] !== $basePageStyleSize[0] || $defaultOptionPaperSize[1] !== $basePageStyleSize[1]
+                || $this->paperOrientation !== $options->getDefaultPaperOrientation()) {
+                $this->setPaper(array(0, 0, $basePageStyleSize[0], $basePageStyleSize[1]), $this->paperOrientation);
```

Comment:
I'm thinking we don't want to use the previously-defined paper orientation here. In the [code that parses the CSS-defined page size](https://github.com/dompdf/dompdf/blob/v0.7.0/src/Css/Style.php#L2238) orientation is already taken into account. Since the purpose of orientation right now is merely to switch the height/width during Canvas construction, and considering that switch has already happened in the style code, we should probably just set it to portrait.